src/openforms/prefill/__init__.py

```diff
@@ -139,11 +145,13 @@ def prefill_variables(submission: "Submission", register=None) -> None:
     state = submission.load_submission_value_variables_state()
     variables_to_prefill = state.get_prefill_variables()
 
-    grouped_fields = defaultdict(list)
+    grouped_fields = defaultdict(dict)
```

Comment:
```suggestion
    grouped_fields = defaultdict(defaultdict(list)
```
and then you'd be able to avoid the `setdefault` call later on?

Reply:
Ahh I forgot that the argument must be a callable